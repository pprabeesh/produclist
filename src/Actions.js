export const actions = {
  add: products => ({
    type: "add",
    productList: products
  }),
  update: filterBy => ({
    type: "update",
    data: filterBy
  }),
  filter: filterOption => ({
    type: "filter",
    selectedOption: filterOption
  })
};
