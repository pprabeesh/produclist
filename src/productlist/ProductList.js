import React, { useEffect, useState } from "react";
import axios from "axios";
import Product from "../product/Product";
import { StyledContainer, Styledflex } from "./ProductList.styles";
import { DATAURL } from "../Constants";

export const ProductList = () => {
  const [products, setProducts] = useState([]);
  useEffect(() => {
    axios.get(DATAURL).then(result => setProducts(result.data));
  }, []);
  return (
    <React.Fragment>
      <StyledContainer>
        <Styledflex>
          {products &&
            products.map(product => (
              <Product key={product.index} {...product} />
            ))}
        </Styledflex>
      </StyledContainer>
    </React.Fragment>
  );
};

export default ProductList;
