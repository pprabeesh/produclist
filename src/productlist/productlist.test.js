import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { actions } from "../Actions";
import ProductList from "./ProductList";
import { FilterContext } from "../Context";

test("ProductList renders correctly", () => {
  const state = {};
  const dispatch = () => {};
  let container = document.createElement("div");

  renderer.act(() => {
    ReactDOM.render(
      <FilterContext.Provider value={{ state, actions, dispatch }}>
        <ProductList />
      </FilterContext.Provider>,
      container
    );
  });

  expect(container).toMatchSnapshot("Not expanded");
});
