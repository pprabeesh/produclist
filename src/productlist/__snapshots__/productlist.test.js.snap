// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders correctly: Not expanded 1`] = `
<div>
  <div
    class="sc-htoDjs bxeJIL"
  >
    <div
      class="sc-bwzfXH sc-dnqmqq clqoMf sc-bdVaJa iHZvIS"
    />
  </div>
</div>
`;
